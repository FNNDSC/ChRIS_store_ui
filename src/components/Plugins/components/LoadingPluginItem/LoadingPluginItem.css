.loading-plugin-item {
  /* display */
  padding-bottom: 20px;

  /* border */
  border-bottom-color: #ededed;
  border-bottom: 4px solid var(--pf-black-200);
}