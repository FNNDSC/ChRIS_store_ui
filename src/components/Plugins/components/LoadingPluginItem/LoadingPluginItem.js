import React from 'react';
import LoadingContainer from '../../../LoadingContainer/LoadingContainer';
import LoadingContent from '../../../LoadingContainer/components/LoadingContent/LoadingContent';
import './LoadingPluginItem.css';

const LoadingPluginItem = () => (
  <LoadingContainer className="loading-plugin-item" >
    <LoadingContent
      width="150px"
      height="37px"
      top="20px"
      left="20px"
    />
    <LoadingContent
      width="160px"
      height="23px"
      top="7.5px"
      left="20px"
    />
    <LoadingContent
      width="230px"
      height="23px"
      top="7.5px"
      left="20px"
    />
  </LoadingContainer>
);

export default LoadingPluginItem;
