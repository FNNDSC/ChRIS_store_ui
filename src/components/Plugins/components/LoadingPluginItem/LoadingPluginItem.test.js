import React from 'react';
import { shallow } from 'enzyme';
import LoadingPluginItem from './LoadingPluginItem';

describe('LoadingPluginItem', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<LoadingPluginItem />);
  });

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render LoadingContainer component', () => {
    expect(wrapper.find('LoadingContainer')).toHaveLength(1);
  });

  it('should render LoadingContainer with loading-plugin-item class', () => {
    expect(wrapper
      .find('LoadingContainer.loading-plugin-item')
      .prop('className'))
      .toEqual('loading-plugin-item');
  });

  it('should render 3 LoadingContent components', () => {
    expect(wrapper.find('LoadingContent')).toHaveLength(3);
  });
});
