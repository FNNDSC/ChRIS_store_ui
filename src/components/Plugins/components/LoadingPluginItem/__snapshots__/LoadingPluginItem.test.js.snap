// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingPluginItem should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LoadingPluginItem />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <LoadingContent
          bottom="0"
          className=""
          height="37px"
          left="20px"
          right="0"
          top="20px"
          type=""
          width="150px"
/>,
        <LoadingContent
          bottom="0"
          className=""
          height="23px"
          left="20px"
          right="0"
          top="7.5px"
          type=""
          width="160px"
/>,
        <LoadingContent
          bottom="0"
          className=""
          height="23px"
          left="20px"
          right="0"
          top="7.5px"
          type=""
          width="230px"
/>,
      ],
      "className": "loading-plugin-item",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "bottom": "0",
          "className": "",
          "height": "37px",
          "left": "20px",
          "right": "0",
          "top": "20px",
          "type": "",
          "width": "150px",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "bottom": "0",
          "className": "",
          "height": "23px",
          "left": "20px",
          "right": "0",
          "top": "7.5px",
          "type": "",
          "width": "160px",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "bottom": "0",
          "className": "",
          "height": "23px",
          "left": "20px",
          "right": "0",
          "top": "7.5px",
          "type": "",
          "width": "230px",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <LoadingContent
            bottom="0"
            className=""
            height="37px"
            left="20px"
            right="0"
            top="20px"
            type=""
            width="150px"
/>,
          <LoadingContent
            bottom="0"
            className=""
            height="23px"
            left="20px"
            right="0"
            top="7.5px"
            type=""
            width="160px"
/>,
          <LoadingContent
            bottom="0"
            className=""
            height="23px"
            left="20px"
            right="0"
            top="7.5px"
            type=""
            width="230px"
/>,
        ],
        "className": "loading-plugin-item",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "bottom": "0",
            "className": "",
            "height": "37px",
            "left": "20px",
            "right": "0",
            "top": "20px",
            "type": "",
            "width": "150px",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "bottom": "0",
            "className": "",
            "height": "23px",
            "left": "20px",
            "right": "0",
            "top": "7.5px",
            "type": "",
            "width": "160px",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "bottom": "0",
            "className": "",
            "height": "23px",
            "left": "20px",
            "right": "0",
            "top": "7.5px",
            "type": "",
            "width": "230px",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
